### DISCUSSION TITLE
Flashing hex file

### DISCUSSION THREAD

**Discussion by prau0418** [ElectroTechnique/electrotechnique.github.io#8] [Software Issues]
I’m a novice and would appreciate a more step by step guide to how to flash the hex file to the teensy board.  Thanks.

**Comment by prau0418** [comment]
I got an answer from Simon.  Download the whole repository and get it from the zip file. Use the Teensy loader application as instructed in the build guide.

**Reply by ericthekraut** [reply]
Where do you find that?

**Reply by Electrotechnique** [reply]
https://github.com/ElectroTechnique/TSynth-Teensy4.1/tree/master/Firmware

**Comment by Electrotechnique** [comment]
Yes, download the entire repository as a zip file and get the hex file from it. Otherwise it'll download as a text file that won't work.

**Comment by ericthekraut** [comment]
That is the exact hex file I uploaded to my teensy. And it still doesn’t work

**Reply by Electrotechnique** [reply]
"It doesn't work" is really insufficient information isn't it? It flashes and doesn't work or it simply won't flash? You need to download the entire repository and then take the hex file, otherwise it will download as text.

**Comment by ericthekraut** [comment]
So I’ve made some progress. I actually got it to make sounds. The problem now, is, that the display is skewed. I’ve made the change to ST7735Display.h. I changed GREENTAB to BLACKTAB but I don’t know how to turn the repository into a hex file. Teensyduino won’t do it, I get an error message.




This is the error message I get. I downloaded circular buffer but I don’t know what to do with it, and I downloaded the adafruit gfx library but I’m not sure what to do with that. Any specific step by step instructions would be most appreciated


> On Oct 30/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"

**Reply by Electrotechnique** [reply]
You need to install Arduino IDE, then TeensyDuino and then install CircularBuffer and AdafruitGFX from the library manager. Someone in the FaceBook group with a similar display can probably help you and provide a compiled hex file:

[https://www.facebook.com/groups/418750379310464](https://www.facebook.com/groups/418750379310464)
